src/Calculator/Views/GraphingCalculator/KeyGraphFeaturesPanel.xaml.cpp

```diff
@@ -0,0 +1,282 @@
+#include "pch.h"
+#include "KeyGraphFeaturesPanel.xaml.h"
+#include "Controls/MathRichEditBox.h"
+
+using namespace CalculatorApp;
+using namespace CalculatorApp::ViewModel;
+using namespace Platform;
+using namespace Platform::Collections;
+using namespace Windows::UI;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+
+KeyGraphFeaturesPanel::KeyGraphFeaturesPanel()
+    : m_Domain{ "" }
+    , m_Range{ "" }
+    , m_XIntercept{ "" }
+    , m_YIntercept{ "" }
+    , m_Parity{ "" }
+    , m_Periodicity{ "" }
+    , m_Minima{ "" }
+    , m_Maxima{ "" }
+    , m_InflectionPoints{ "" }
+    , m_Monotonicity{ "" }
+    , m_VerticalAsymptotes{ "" }
+    , m_HorizontalAsymptotes{ "" }
+    , m_ObliqueAsymptotes{ "" }
+    , m_resourceLoader{ Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView() }
+{
+    InitializeComponent();
+}
+void KeyGraphFeaturesPanel::OnPropertyChanged(String ^ propertyName)
+{
+    if (propertyName == L"ViewModel")
+    {
+        if (ViewModel == nullptr)
+        {
+            return;
+        }
+        if (ViewModel->AnalysisNotSupported)
+        {
+            AnalysisNotSupported = m_resourceLoader->GetString(L"KGFAnalysisNotSupported");
+            AnalysisNotSupportedTextBlock->Visibility = Windows::UI::Xaml::Visibility::Visible;
+            KeyGraphFeaturesStackPanel->Visibility = Windows::UI::Xaml::Visibility::Collapsed;
+        }
+        else
+        {
+            AnalysisNotSupportedTextBlock->Visibility = Windows::UI::Xaml::Visibility::Collapsed;
+            KeyGraphFeaturesStackPanel->Visibility = Windows::UI::Xaml::Visibility::Visible;
+
+            DomainTextBlock->Text = m_resourceLoader->GetString(L"Domain");
+            RangeTextBlock->Text = m_resourceLoader->GetString(L"Range");
+            XInterceptTextBlock->Text = m_resourceLoader->GetString(L"XIntercept");
+            YInterceptTextBlock->Text = m_resourceLoader->GetString(L"YIntercept");
+            ParityTextBlock->Text = m_resourceLoader->GetString(L"Parity");
+            PeriodicityTextBlock->Text = m_resourceLoader->GetString(L"Periodicity");
+            MinimaTextBlock->Text = m_resourceLoader->GetString(L"Minima");
+            MaximaTextBlock->Text = m_resourceLoader->GetString(L"Maxima");
+            InflectionPointsTextBlock->Text = m_resourceLoader->GetString(L"InflectionPoints");
+            VerticalAsymptotesTextBlock->Text = m_resourceLoader->GetString(L"VerticalAsymptotes");
+            HorizontalAsymptotesTextBlock->Text = m_resourceLoader->GetString(L"HorizontalAsymptotes");
+            ObliqueAsymptotesTextBlock->Text = m_resourceLoader->GetString(L"ObliqueAsymptotes");
+            MonotonicityTextBlock->Text = m_resourceLoader->GetString(L"Monotonicity");
+
+            Domain = ViewModel->Domain;
```

Comment:
No I didn't directly bind the viewmodel to the UI so that I could substitute a the value with a string if nothing was returned for that property.

Reply:
Since we can show the information directly from the engine or a string provided by the UI, I decided not to directly bind to the xaml so that I could swap out the data more easily. Domain and Range or the only two that do not have an error string associated with them, so I decided not to bind them to the xaml directly just to be consistent with the other controls.

The xaml binding is to the properties in the KeyGraphFeaturesPanel.xaml.h file.

Reply:
Just saw your next comment, I'll move the logic to the viewmodel and bind directly. Thanks for the suggestion.